### PR TITLE
Fix `declaration-block-no-duplicate-properties` false positives with option `consecutive-duplicates-with-different-syntaxes`

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/6796

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self-explanatory."

### CODE REVIEW

File: lib/rules/declaration-block-no-duplicate-properties/index.js [hunk 1]
```diff
@@ -38,34 +39,47 @@ const isEqualValueNodes = (nodes1, nodes2) => {
...
+
+		// Custom properties have unknown value syntaxes but are equal for CSS parsers.
+		if (
+			node1.type === 'Identifier' &&
+			node1Name.startsWith('--') &&
+			node2.type === 'Identifier' &&
+			node2Name.startsWith('--')
+		) {
+			continue;
+		}
```
Comment (romainmenke) [inline lib/rules/declaration-block-no-duplicate-properties/index.js:58]: Dashed idents are different from regular idents.
These are always known to be the same syntax.

File: .changeset/many-toes-smell.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed false positives with option `ignore: ["consecutive-duplicates-with-different-syntaxes"]` in `declaration-block-no-duplicate-properties`
```
Comment (ybiquitous) [inline .changeset/many-toes-smell.md:5]: [suggest] Can you fix the message for consistency?

```suggestion
Fixed: `declaration-block-no-duplicate-properties` false positives with option `ignore: ["consecutive-duplicates-with-different-syntaxes"]`
```

File: lib/rules/declaration-block-no-duplicate-properties/index.js [hunk 3]
```diff
@@ -38,34 +39,47 @@ const isEqualValueNodes = (nodes1, nodes2) => {
...
 
+		const node1Name = 'name' in node1 ? String(node1.name) : '';
+		const node2Name = 'name' in node2 ? String(node2.name) : '';
+
+		// Custom properties have unknown value syntaxes but are equal for CSS parsers.
+		if (
+			node1.type === 'Identifier' &&
+			node1Name.startsWith('--') &&
+			node2.type === 'Identifier' &&
+			node2Name.startsWith('--')
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-duplicate-properties/index.js:55]: [suggest] Can you use `isCustomProperty()` for consistency?

```suggestion
			isCustomProperty(node1Name) &&
			node2.type === 'Identifier' &&
			isCustomProperty(node2Name)
```

Comment (romainmenke) [reply lib/rules/declaration-block-no-duplicate-properties/index.js:55]: I should have checked what this function does :)
I assumed it took a declaration as its argument.

**Review by ybiquitous — APPROVED** [review]
Thanks, LGTM 👍🏼

**Comment by romainmenke** [comment]
Thank you for reviewing 🙇

**Pull request merged**